migrations/20221011041400_add_mfa_indexes.up.sql

```diff
@@ -1,8 +1,27 @@
-create index if not exists refresh_token_session_id on {{ index .Options "Namespace" }}.refresh_tokens using btree(session_id);
+-- Taken from: https://stackoverflow.com/questions/6801919/postgres-add-constraint-if-it-doesnt-already-exist
+create or replace function create_auth_constraint_if_not_exists (
+    c_name text, constraint_sql text
+)
+returns void AS
+$$
+begin
+    -- Modify only auth constraint
+    if not exists ( select constraint_name
+                from    information_schema.check_constraints
+                where   constraint_schema = 'auth'
+                  and   constraint_name = c_name
+              )
+    then
+        execute constraint_sql;
+    end if;
+end;
+$$
+language 'plpgsql';
```

Comment:
hmm i don't think we need to create a function in the auth schema just to handle this?

what about following what we did here for adding a constraint to the sessions table: https://github.com/supabase/gotrue/blob/master/migrations/20220811173540_add_sessions_table.up.sql#L15-L23

Reply:
Yeah, you're right - an additional function might be confusing for users - was hoping to reuse in the case that we need to add additional constraints in the future. [information_schema.constraint_column_usage](https://github.com/supabase/gotrue/blob/master/migrations/20220811173540_add_sessions_table.up.sql#L15-L23) could work as well but have a slight preference of using `information_schema.check_constraints` as it doesn't require the table to be specified so will go with that unless there are additional considerations that I've missed